3 урок

KatPierce opened the pull request.

KatPierce committed:
task1 all

KatPierce committed:
task1 all исправленное

KatPierce committed:
task2 all

KatPierce committed:
task1 all

KatPierce committed:
task1 all

KatPierce committed:
task1/2

KatPierce committed:
task3

KatPierce committed:
task3

KatPierce committed:
task3

KatPierce committed:
task3

KatPierce committed:
task3

belyaev-mikhail commented:
recheck all

belyaev-mikhail commented:
recheck all

belyaev-mikhail commented on src/lesson1/task1/Simple.kt:
```diff
@@ -93,7 +101,7 @@ fun thirdDigit(number: Int): Int = TODO()
  * Определите время поезда в пути в минутах (в данном случае 216).
  */
-fun travelMinutes(hoursDepart: Int, minutesDepart: Int, hoursArrive: Int, minutesArrive: Int): Int = TODO()
+fun travelMinutes(hoursDepart: Int, minutesDepart: Int, hoursArrive: Int, minutesArrive: Int): Int = (hoursArrive * 60 + minutesArrive) - (hoursDepart * 60 + minutesDepart)
```
Длинные строки в коде лучше разбивать на более короткие

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,126 @@ fun digitCountInNumber(n: Int, m: Int): Int =
-fun maxDivisor(n: Int): Int = TODO()
-
+fun isCoPrime(m: Int, n: Int): Boolean {
+    val a = Math.max(m, n)
```
Эту функцию гораздо проще (и быстрее) можно реализовать на основе функций `gcd` или `lcm`, определённых выше.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,126 @@ fun digitCountInNumber(n: Int, m: Int): Int =
+fun maxDivisor(n: Int): Int {
+    var divisor = 2
+    if (isPrime(n) == true) return 1
+    else for (i in 2..n - 1) {
```
Действительно ли для проверки делителей нужно проверять все числа до `n - 1`?
Как минимум, _половина_ из них делителями `n` точно не является.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,126 @@ fun digitCountInNumber(n: Int, m: Int): Int =
  */
-fun squareBetweenExists(m: Int, n: Int): Boolean = TODO()
+// Программа проверяет, является ли число точным квадратом.
+fun square(m: Int): Boolean {
```
Что это за чушь? Эта программа ничего не проверяет.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,126 @@ fun digitCountInNumber(n: Int, m: Int): Int =
+}
+
+fun squareBetweenExists(m: Int, n: Int): Boolean =
+        (m <= sqr(Math.sqrt(n.toDouble()).toInt().toDouble()) && sqr(Math.sqrt(n.toDouble()).toInt().toDouble()) <= n)
```
1. Вы считает одно и тоже сложное выражение дважды
2. Это некорректный способ проверки, является ли число квадратом. В том случае, если `n == m` и оно само по себе квадрат, может произойти ошибка округления и функция выдаст `false`. Либо используйте функции округления вместо `toInt()`, либо не используйте такой способ.
3. `sqr(Math.sqrt(n.toDouble()).toInt().toDouble()) <= n` является верным вообще всегда для любых `n`, кроме случаев с ошибками округления.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -157,7 +261,29 @@ fun hasDifferentDigits(n: Int): Boolean = TODO()
+    var result=0
+    while (n>count) {
+        number= i*i
+        nn= number
```
Пробелы

KatPierce committed:
все и сразу(1,3,4)

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -104,31 +106,58 @@ fun buildSumExample(list: List<Int>) = list.joinToString(separator = " + ", post
+    val n = v.size
+    var result = 0.0
+    for (i in 0..n-1){
+        result+=v[i]*v[i]}
```
Пробелы

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -104,31 +106,58 @@ fun buildSumExample(list: List<Int>) = list.joinToString(separator = " + ", post
  */
-fun mean(list: List<Double>): Double = TODO()
+fun mean(list: List<Double>): Double {
+   val sum= list.sum()
```
Отступы и пробелы

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -104,31 +106,58 @@ fun buildSumExample(list: List<Int>) = list.joinToString(separator = " + ", post
+    if(list.size==0) return list else
+    for (i in 0..list.size - 1) {
+        val element = list[i]
+            list[i] = element-n
```
Некорректные отступы

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -104,31 +106,58 @@ fun buildSumExample(list: List<Int>) = list.joinToString(separator = " + ", post
+    val n= a.size
+    var result = 0.0
+    if ((a.size==0)&&(b.size==0)) return 0.0 else
+        for (i in 0..n-1){
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -148,7 +188,17 @@ fun polynom(p: List<Double>, x: Double): Double = TODO()
+    var sum=0.0
+    val n= list.size
+    if(list.isEmpty()==true) return list else
+         for (i in 0..n - 1) {
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -157,15 +207,31 @@ fun accumulate(list: MutableList<Double>): MutableList<Double> = TODO()
+    while (number !=1) {
+        element = minDivisor(number)
+        result.add(element)
+        number/=element
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -157,15 +207,31 @@ fun accumulate(list: MutableList<Double>): MutableList<Double> = TODO()
-fun factorizeToString(n: Int): String = TODO()
+fun factorizeToString(n: Int): String {
+    val list = factorize(n)
+    var a =""
```
Почему `var`, если переменная не меняется?

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -157,15 +207,31 @@ fun accumulate(list: MutableList<Double>): MutableList<Double> = TODO()
+    val list = factorize(n)
+    var a =""
+    a= list.joinToString(separator = "*")
+    if (isPrime(n)==true ) return "$n" else return a
```
Что это значит и зачем это? `factorize` и так должен вернуть список из одного `n`, если `n` &mdash; простое.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,111 @@ fun digitCountInNumber(n: Int, m: Int): Int =
+fun sqr(x: Int) = x * x
+
+fun squareBetweenExists(m: Int, n: Int): Boolean =
+        m <= sqr(Math.sqrt(n.toDouble()).toInt())
```
Это по-прежнему некорректный код для вычисления ближайшего снизу квадрата.

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -157,7 +246,30 @@ fun hasDifferentDigits(n: Int): Boolean = TODO()
+    dif = count - n
+    if (n == count) result = nn % 10
+    else
+        result = (nn / Math.pow(10.0, dif.toDouble()).toInt()) % 10
```
Так возводить в степень целые числа нельзя.

mglukhikh commented:
Вы довольно сильно отстаёте от графика. Вам следует продолжать решать задачи учебного проекта.

Kazachenko1998 committed:
исправленнное в 3 и  4 уроке

KatPierce commented:
У меня полетел компьютер, отправляю с ноутбука Константина Казаченко.

Kazachenko1998 committed:
task6

Kazachenko1998 committed:
task5

Kazachenko1998 committed:
task5

Kazachenko1998 committed:
task5

Kazachenko1998 committed:
task5

Kazachenko1998 committed:
task5

Kazachenko1998 committed:
task6 2

belyaev-mikhail commented:
recheck all

belyaev-mikhail commented:
recheck all

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,55 +60,117 @@ fun digitCountInNumber(n: Int, m: Int): Int =
+fun squareBetweenExists(m: Int, n: Int): Boolean {
+    for (result in m..n) {
+        val sqrtResult = Math.sqrt(result.toDouble())
+        if (sqrtResult % 1 == 0.0) {
```
И эта проверка по-прежнему некорректна.

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -128,7 +152,16 @@ fun center(list: MutableList<Double>): MutableList<Double> = TODO()
+fun times(a: List<Double>, b: List<Double>): Double {
+    val n = a.size
+    var result = 0.0
+    if ((a.size == 0) && (b.size == 0)) return 0.0 else
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -138,7 +171,17 @@ fun times(a: List<Double>, b: List<Double>): Double = TODO()
+fun polynom(p: List<Double>, x: Double): Double {
+    val n = p.size
+    var result = 0.0
+    if (p.isEmpty() == true) return 0.0 else
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -148,7 +191,17 @@ fun polynom(p: List<Double>, x: Double): Double = TODO()
+fun accumulate(list: MutableList<Double>): MutableList<Double> {
+    var sum = 0.0
+    val n = list.size
+    if (list.isEmpty() == true) return list else
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -184,7 +265,16 @@ fun convert(n: Int, base: Int): List<Int> = TODO()
+    var result = ""
+    list = convert(n, base)
+    for (i in 0..list.size - 1) {
+        if (list[i] > 9) result += (87 + list[i]).toChar() else
```
Форматирование

belyaev-mikhail commented on src/lesson4/task1/List.kt:
```diff
@@ -204,7 +302,16 @@ fun decimal(digits: List<Int>, base: Int): Int = TODO()
+    var list: List<Int>
+    list = listOf()
+    for (i in 0..str.length - 1) {
+        if (str[i] in '0'..'9') list += ((str[i]).toInt() - 48) else
```
Форматирование

belyaev-mikhail commented on src/lesson6/task1/Geometry.kt:
```diff
@@ -55,14 +55,18 @@ data class Circle(val center: Point, val radius: Double) {
      */
-    fun contains(p: Point): Boolean = TODO()
+    fun contains(p: Point): Boolean =
+            (sqr(center.x-p.x)+sqr(center.y-p.y))<=sqr(radius)
```
Отступы

belyaev-mikhail commented on src/lesson6/task1/Geometry.kt:
```diff
@@ -76,15 +80,29 @@ data class Segment(val begin: Point, val end: Point)
  */
-fun circleByDiameter(diameter: Segment): Circle = TODO()
+fun circleByDiameter(diameter: Segment): Circle =  Circle(center = Point((diameter.begin.x + diameter.end.x) / 2.0, (diameter.begin.y + diameter.end.y) / 2.0)
+        , radius = diameter.begin.distance(diameter.end) / 2.0)
```
Кривое форматирование

belyaev-mikhail commented on src/lesson6/task1/Geometry.kt:
```diff
@@ -105,7 +123,7 @@ data class Line(val point: Point, val angle: Double) {
  * Построить прямую по отрезку
  */
-fun lineBySegment(s: Segment): Line = TODO()
+fun lineBySegment(s: Segment): Line = Line(s.begin, Math.atan((s.end.y-s.begin.y)/(s.end.x-s.begin.x)))
```
Пробелы

Commit:
6,7

mglukhikh commented on src/lesson5/task1/Parse.kt:
```diff
@@ -187,4 +286,142 @@ fun fromRoman(roman: String): Int = TODO()
+}
+
+
+fun exam(examResults: List<String>, thresherhold: Double): List<String> {
```
Удалите эту функцию из репозитория, пожалуйста

mglukhikh commented on src/lesson5/task1/Parse.kt:
```diff
@@ -187,4 +286,142 @@ fun fromRoman(roman: String): Int = TODO()
+    return resultList
+}
+
+fun beststudents(examResults: List<String>, threshold: Double): List<String> {
```
Удалите эту функцию из репозитория, пожалуйста

mglukhikh commented on src/lesson5/task1/Parse.kt:
```diff
@@ -187,4 +286,142 @@ fun fromRoman(roman: String): Int = TODO()
+    return result
+}
+
+fun student(examResults: List<String>, threshold: Double): List<String> {
```
Удалите эту функцию из репозитория, пожалуйста

mglukhikh commented on src/lesson5/task1/Parse.kt:
```diff
@@ -187,4 +286,142 @@ fun fromRoman(roman: String): Int = TODO()
+    return result
+}
+
+fun birthdays(text: String): List<String> {
```
Удалите эту функцию из репозитория, пожалуйста

mglukhikh reviewed (changes requested):
Большая просьба не оставлять в репозитории решений задач с тестирования. А ещё лучше было бы туда их и вовсе не добавлять

KatPierce commented:
Михаил Игоревич, я не понимаю. У меня получилось отправить пуллреквест или
нет? Вчера мы с Петровым Максимом Алексеевичем 2 часа пытались это сделать
,в итоге он отправился, но бот прислал вот это.

пт, 23 дек. 2016 г. в 9:48, Mikhail Glukhikh <notifications@github.com>:

> *@mglukhikh* requested changes on this pull request.
>
>
>
>
> Большая просьба не оставлять в репозитории решений задач с тестирования. А
> ещё лучше было бы туда их и вовсе не добавлять
> ------------------------------
>
>
>
>
> In src/lesson5/task1/Parse.kt
> <https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193#pullrequestreview-14267560>
> :
>
>
> > \ No newline at end of file
>
> +fun computeDeviceCells(cells: Int, commands: String): List<Int> = TODO()
>
> +
>
> +fun main(args: Array<String>) {
>
> +    val f = "Он пошёл в в школу"
>
> +    val x = firstDuplicateIndex(f)
>
> +    println(x)
>
> +}
>
> +
>
> +
>
> +fun exam(examResults: List<String>, thresherhold: Double): List<String> {
>
>
>
> Удалите эту функцию из репозитория, пожалуйста
>
>
>
>
> ------------------------------
>
>
>
>
> In src/lesson5/task1/Parse.kt
> <https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193#pullrequestreview-14267560>
> :
>
>
> > +            try {
>
> +                sr += stringLittleLittle[2].toString().toDouble()
>
> +            } catch (e: NumberFormatException) {
>
> +                throw IllegalArgumentException()
>
> +            }
>
> +
>
> +            number++
>
> +        }
>
> +        if (sr / number >= thresherhold) {
>
> +            resultList += (string[0].substring(0, string[0].length - 1))
>
> +        }
>
> +    }
>
> +    return resultList
>
> +}
>
> +
>
> +fun beststudents(examResults: List<String>, threshold: Double): List<String> {
>
>
>
> Удалите эту функцию из репозитория, пожалуйста
>
>
>
>
> ------------------------------
>
>
>
>
> In src/lesson5/task1/Parse.kt
> <https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193#pullrequestreview-14267560>
> :
>
>
> > +        var mark = 0.0
>
> +        var marks = listOf<String>()
>
> +        for (j in 0..part.size - 1) {
>
> +            marks = part[j].split(" ")
>
> +            mark = marks[1].toDouble()
>
> +            if (mark < 2.0 || mark > 5.0)
>
> +                throw IllegalArgumentException()
>
> +            averagemark += mark
>
> +        }
>
> +        if (averagemark / part.size >= threshold)
>
> +            result.add(parse[0])
>
> +    }
>
> +    return result
>
> +}
>
> +
>
> +fun student(examResults: List<String>, threshold: Double): List<String> {
>
>
>
> Удалите эту функцию из репозитория, пожалуйста
>
>
>
>
> ------------------------------
>
>
>
>
> In src/lesson5/task1/Parse.kt
> <https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193#pullrequestreview-14267560>
> :
>
>
> > +        var mark = 0.0
>
> +        var marks = listOf<String>()
>
> +        for (j in 0..part.size - 1) {
>
> +            marks = part[j].split(" ")
>
> +            mark = marks[1].toDouble()
>
> +            if (mark < 2.0 || mark > 5.0)
>
> +                throw IllegalArgumentException()
>
> +            averagemark += mark
>
> +        }
>
> +        if (averagemark / part.size >= threshold)
>
> +            result.add(parse[0])
>
> +    }
>
> +    return result
>
> +}
>
> +
>
> +fun birthdays(text: String): List<String> {
>
>
>
> Удалите эту функцию из репозитория, пожалуйста
>
>
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193#pullrequestreview-14267560>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVLQ_05MuwfLxgtloH1lUuhB_5MpEsjTks5rK26vgaJpZM4KhUzt>
> .
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>

mglukhikh commented:
Всё у вас получилось. Это не бот прислал, это мой комментарий. Потому что не нужно класть в репозиторий решения задач с тестирования, ещё раз вам повторяю. Вообще, комментарии и результаты смотреть лучше на страничке вашего PR: https://github.com/Kotlin-Polytech/KotlinAsFirst2016/pull/193.
